src/bumper.cpp

```diff
@@ -0,0 +1,103 @@
+#include "bumper.h"
+
+Bumper::Bumper()
+{
+}
+
+Bumper::~Bumper()
+{
+
+}
+
+void Bumper::setRobotRadius(double width, double length)
+{
+    _robotRadiusWidth = width;
+    _robotRadiusLength = length;
+}
+
+void Bumper::setRobotRadius(double radius)
+{
+    _robotRadiusLength = radius;
+    _robotRadiusWidth = radius;
+}
+
+void Bumper::setBumperRadius(double size)
+{
+    _bumperRadius = size;
+}
+
+void Bumper::generateBumperData(const World& world, const Robot& robot, std_msgs::Bool& scan_msg_f, std_msgs::Bool& scan_msg_r) const
+{
+    // Generate laser data to perform bumper check
+    sensor_msgs::LaserScan lrf_msg;
+    LRF::generateLaserData(world, robot, lrf_msg);
+
+    bool hitF = false;
+    bool hitR = false;
+
+    int N = lrf_msg.ranges.size();
+    for(int i = 0; i<lrf_msg.ranges.size(); i++)
```

Comment:
Je weet dat de eerste 1/4 van je meetpunten op de achterkant liggen. De middelste helft aan de voorkant en de laatste 1/4 weer aan de achterkant. Maak daar gebruik van

Reply:
Uhm ja kan. Wat stel je voor? Twee/drie for-loopjes en die afbreken als je  er een hit gevonden hebt?

Reply:
Precies. Je kan ook het bereik van je LRF aanpassen naar -0.5pi tot 1.5 pi zodat je maar twee for loops hoeft te gebruiken: een voor voor en een voor achter.

Reply:
Ja oke eens.